### PR TITLE
[Debt] Migrate pool operations

### LINKED ISSUE #8717: ♻️ Migrate All Pool Items to `client-preset`
## ♻️ Debt/Refactor

Migrate all `*.operations.graphql` files related to `Pool` and `PoolCandidate`  and associated components to use the new pattern in `@gc-digital-talent/graphql`

## 🕵️ Details

Follow [the documentation](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/documentation/graphql.md) to complete this issue.

## ✅ Acceptance Criteria

- [ ] `*.graphql` files removed
- [ ] All components used by `Pool` and `PoolCandidate` use new pattern
- [ ] All components used by `Pool` and `PoolCandidate` are fragmentized
- [ ] Cypress tests updated to use new queries
- [ ] Refactor usage of `Maybe` to exclude `undefined`

### PR DESCRIPTION
**Pull request by JamesHuf**
🤖 Resolves #8717.

## 👋 Introduction

This PR will migrate the pool and poolCandidate workflows to the new graphql pattern.

## 🧪 Testing

1. Build `npm run dev`
2. Test the various pages focused on pools and poolCandidates (admin, browse pools, search page)
3. Confirm functionality remains intact
4. Look for any other operations it would make sense to update with this PR.

### CODE REVIEW

File: apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx [hunk 1]
```diff
@@ -20,12 +20,12 @@ import {
   getPoolCandidateStatus,
 } from "@gc-digital-talent/i18n";
 import { toast } from "@gc-digital-talent/toast";
+import { graphql } from "@gc-digital-talent/graphql/src/gql/gql";
```
[the remark this answers is not in the document]
Comment (JamesHuf) [reply apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx]: Oops fixed in [Update imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8973/commits/e6578b12584a7f3628c2b5cd1b340a8da5d02749)

File: apps/web/src/components/PoolCandidatesTable/SkillMatchDialog.tsx [hunk 2]
```diff
@@ -1,17 +1,86 @@
...
 import { notEmpty } from "@gc-digital-talent/helpers";
+import { Maybe, Skill, graphql } from "@gc-digital-talent/graphql";
 
-import {
-  Maybe,
-  Skill,
-  Scalars,
-  useGetSkillMatchDialogDataQuery,
-} from "~/api/generated";
+import { Scalars } from "~/api/generated";
```
[the remark this answers is not in the document]
Comment (JamesHuf) [reply apps/web/src/components/PoolCandidatesTable/SkillMatchDialog.tsx]: Fixed in [Update imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8973/commits/e6578b12584a7f3628c2b5cd1b340a8da5d02749)

File: apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/ViewPoolCandidatePage.tsx [hunk 3]
```diff
@@ -21,16 +22,15 @@ import {
...
-  Scalars,
-  useGetPoolCandidateSnapshotQuery,
-  PoolCandidate,
   Maybe,
   SkillCategory,
-  Pool,
-} from "~/api/generated";
+} from "@gc-digital-talent/graphql";
+
+import { Scalars } from "~/api/generated";
```
[the remark this answers is not in the document]
Comment (JamesHuf) [reply apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/ViewPoolCandidatePage.tsx]: Fixed in [Update imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8973/commits/e6578b12584a7f3628c2b5cd1b340a8da5d02749)

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx [hunk 4]
```diff
@@ -37,16 +37,123 @@ const getFlourishStyles = (isTop: boolean) => ({
   "data-h2-z-index": "base(-1)",
 });
 
-export interface BrowsePoolsProps {
-  pools: Pool[];
-}
+// export interface BrowsePoolsProps {
+//   pools: Pool[];
+// }
```
[the remark this answers is not in the document]
Comment (JamesHuf) [reply apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx]: Forgot about this. ✂️ in [Remove commented code](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8973/commits/d66ddb4175ded205a77a7e71882c251fdfd6c39f)

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx [hunk 5]
```diff
@@ -270,26 +380,25 @@ export const BrowsePools = ({ pools }: BrowsePoolsProps) => {
...
-      <BrowsePools pools={filteredPools} />
-    </Pending>
-  );
-};
+//   return (
+//     <Pending fetching={fetching} error={error}>
+//       <BrowsePools pools={filteredPools} />
+//     </Pending>
+//   );
+// };
```
[the remark this answers is not in the document]
Comment (JamesHuf) [reply apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx]: Forgot about this. ✂️ in [Remove commented code](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8973/commits/d66ddb4175ded205a77a7e71882c251fdfd6c39f)

**Review by esizer — APPROVED** [review]
Awesome! ⭐

**Pull request merged**